calculator/src/components/Button.js

```diff
@@ -0,0 +1,149 @@
+import { Component } from "react";
+
+
+class Button extends Component{
+    constructor(){
+        super();
+        this.state = {
+            numInput: "",
+            operator: "",
+            lastNum: ""
+        }
+    }
+
+    handleClick = (event) =>{
+        const { numInput } = this.state;
+        this.setState({
+            numInput: numInput + event.target.value
+        });
+    };
+
+    handleAllClear = () =>{
+        this.setState({
+            numInput: "",
+            operator: "",
+            lastNum: ""
+        });
+    };
+
+    handleOperator = (event) => {
```

Comment:
I noticed that when you click an operator button, your input resets to 0, but you can continue to make the calculation and it'll be correct. Just so you know, that's happening because you are setting the numInput to "" every time you click an operator button.